doc/specs/#1012 - Show dependencies.md

```diff
@@ -0,0 +1,111 @@
+
+* Windows Features
+* Windows Libraries
+* Package Dependencies (same source)
+* External Dependencies
+
+Only dependencies declared in the manifest will be shown, not the entire dependency graph.
+
+### install
+The install command will enumerate the dependencies of the package version being installed as follows:
+```
+> winget install Notepad++
+Found Notepad++ [Notepad++.Notepad++]
+This application is licensed to you by its owner.
+Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
+This package requires the following dependencies:
+  - Windows Feature: Hyper-V
+  - Package: Microsot.WindowsTerminal
+Downloading https://github.com/notepad-plus-plus/notepad-plus-plus/releases/download/v7.9.5/npp.7.9.5.Installer.x64.exe
+Successfully verified installer hash
+Starting package install...
+```
+
+### show
+The show command will enumerate the dependencies of the package as follows:
+```
+> winget show Notepad++
+Found Notepad++ [Notepad++.Notepad++]
+Version: 7.9.5
+Publisher: Notepad++ Team
+Author: Don Ho
+Moniker: notepad++
+Description: Notepad++ is a free (as in “free speech” and also as in “free beer”) source code editor and Notepad replacement that supports several languages. Running in the MS Windows environment, its use is governed by GNU General Public License.
+Homepage: https://notepad-plus-plus.org/
+License: GPL-2.0-only
+License Url: https://raw.githubusercontent.com/notepad-plus-plus/notepad-plus-plus/v7.9.5/LICENSE
+Installer:
+  Type: Nullsoft
+  Locale: en-US
+  Download Url: https://github.com/notepad-plus-plus/notepad-plus-plus/releases/download/v7.9.5/npp.7.9.5.Installer.x64.exe
+  SHA256: 4881548cd86491b453520e83c19292c93b9c6ce485a1f9eb9301e3913a9baced
+  Dependencies:
+    - Windows Feature: Hyper-V
+    - Package: Microsot.WindowsTerminal
+```
+
+### upgrade
+The upgrade command will enumerate the dependencies of the package as follows:
+```
+> winget upgrade Notepad++
+Found Notepad++ [Notepad++.Notepad++]
+This application is licensed to you by its owner.
+Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
+This package requires the following dependencies:
+  - Windows Feature: Hyper-V
+  - Package: Microsot.WindowsTerminal
+Successfully verified installer hash
+Starting package install...
+```
+As of now, it will not try to validate nor install any of the dependencies for any package version.
+
+### uninstall
+```
+> winget uninstall Notepad++
+Found Notepad++ [Notepad++.Notepad++]
+Package had dependencies that may not be needed anymore:
+  - Windows Feature: Hyper-V
+  - Package: Microsot.WindowsTerminal
+Starting package uninstall...
+Successfully uninstalled
+```
+Dependencies will not be uninstalled.
+
+### validate
+For this step dependencies will not be validated. 

```

Comment:
I think that validate should attempt to resolve the dependencies against the repositories available to see if they exist or not, it'd save on typos and moderation headaches.

Reply:
@jedieaston just a heads-up. We're not planning on supporting cross-repository dependencies yet. There would only be "one" source to look for dependencies (the same one the manifest came from). The "-m" option would allow you to specify the source to check for those dependencies when you are testing a local manifest.

Reply:
I guess this could be something like:

 `winget validate -m <pathToManifest> -s winget`

Validation could at least (in theory) check to see if the PackageIdentifier values for the dependencies exist at validation time for the given source.

Reply:
Yes, I agree that `validate` doing what checks it can is a good idea.  This iteration of the feature is just acknowledging that dependencies exist by informing the user though, so at most I think we would do something similar to `show` w/r/t dependencies.

@fzanollo is ramping up here so we are taking a very iterative approach.

Reply:
Yes, thank you John.
I thought about informing dependencies on validate but since it doesn't show any manifest information wasn't sure how to go about it. It may be weird to show only dependencies info and not package name or else. What do you think?

Reply:
Validate will show dependencies of all installers found.